docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,409 @@
+
+The Gatsbyjs.org site has a handful of components that have been developed to facilitate writing new content for the blog and the docs. There are also components that help organize and lay out content in various pages across the website.
+
+This guide lists what components are available and explains how to use them. You can also refer to the [code for this page on GitHub](https://github.com/gatsbyjs/gatsby/blob/master/docs/contributing/docs-and-website-components.md) to see to how each component can be used, because they are all embedded here!
+
+Information about authoring in Markdown and styling components on the site is also listed.
+
+## Globally available components
+
+A variety of components have been written to help with formatting code and content across the blog and the docs and don't need to be imported.
+
+---
+
+### Guide List
+
+The `<GuideList />` is a component that renders a list of child docs nested below the current doc in the site's information architecture. It is often used on overview pages like the [Headless CMS](/docs/headless-cms/) guide where many other pages are nested below it, to show what a docs section contains.
+
+#### Usage
+
+The Guide List component takes one prop:
+
+- `slug` - the value of which is already available on every page's context on Gatsbyjs.org by default
+
+The component can be used like this:
+
+```markdown:title=docs/headless-cms
+---
+title: Headless CMS
+---
+
+Overview information about headless CMSs...
+
+<GuideList slug={props.slug} /> // highlight-line
+```
+
+#### Sample
+
+When rendered in a page, the Guide List looks like this:
+
+<GuideList slug="/docs/headless-cms/" />
+
+---
+
+### Egghead Embed
+
+To embed video content from [Egghead](https://egghead.io) on the site, there is an `<EggheadEmbed />` component that adds an iframe with the video inline with other content.
+
+#### Usage
+
+The Egghead Embed component takes two props:
+
+- `lessonLink` - the URL of the lesson from Egghead
+- `lessonTitle` - the name of the lesson that is used to add more information to the embedded iframe.
+
+It can be used like this:
+
+```markdown:title=docs/using-gatsby-image
+---
+title: Using Gatsby Image
+---
+
+This is how you use `gatsby-image`...
+
+<!-- highlight-start -->
+
+<EggheadEmbed
+  lessonLink="https://egghead.io/lessons/gatsby-use-gatsby-image-s-graphql-fragments-for-blurred-up-and-traced-svg-images"
+  lessonTitle="Use gatsby-image's GraphQL fragments for blurred-up and traced SVG images"
+/>
+
+<!-- highlight-end -->
+```
+
+#### Sample
+
+Rendered, it looks like this:
+
+<EggheadEmbed
+  lessonLink="https://egghead.io/lessons/gatsby-use-gatsby-image-s-graphql-fragments-for-blurred-up-and-traced-svg-images"
+  lessonTitle="Use gatsby-image's GraphQL fragments for blurred-up and traced SVG images"
+/>
+
+---
+
+### Pull Quote
+
+The `<Pullquote />` component is used to call out a quote in the blog. It applies borders and styles that make a section of the content pop out to readers.
```

Comment:
Is this component available everywhere? I'm wondering if this should be listed here or in the "other available components" section.

Reply:
It is technically available everywhere (ie it's used in this file without being imported), but we only ever use it in the blog.